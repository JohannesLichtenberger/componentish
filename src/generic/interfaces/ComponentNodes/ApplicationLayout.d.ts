import { ApplicationStyleProperty } from "./ApplicationPage";
import { GenericNode } from "./GenericNode";

export declare type LayoutType = "internal" | "external";
export interface ApplicationLayoutNode extends GenericNode {
    class: never;
    type: LayoutType;
    path?: string;
    styles?: ApplicationStyleProperty;
}
