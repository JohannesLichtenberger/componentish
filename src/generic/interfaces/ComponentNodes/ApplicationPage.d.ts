import { GenericNode } from "./GenericNode";
export interface ApplicationPageNode extends GenericNode {
    children: GenericNode[];
}
