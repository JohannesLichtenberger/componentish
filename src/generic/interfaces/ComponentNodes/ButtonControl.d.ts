import { RouteNode } from "../RouteNode";
import { GenericNode } from "./GenericNode";
export interface ButtonControlNode extends GenericNode {
    children: never;
    route: RouteNode;
}
