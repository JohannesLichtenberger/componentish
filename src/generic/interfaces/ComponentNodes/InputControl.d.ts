import { RouteNode } from "../RouteNode";
import { GenericNode } from "./GenericNode";
export interface InputControlNode extends GenericNode {
    children: never;
    value: string;
    placeHolder?: string;
    type?: inputTypes;
}

export declare type inputTypes = "text" | "mail" | "number" | "tel";
