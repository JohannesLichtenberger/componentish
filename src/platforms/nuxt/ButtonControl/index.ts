import { staticImplements } from "../../../generic/decorators/staticImplements";
import { ButtonControlNode } from "../../../generic/interfaces/ComponentNodes/ButtonControl";
import { GenericNodeTransformer } from "../../../generic/interfaces/transformer/GenericNodeTransformer";

@staticImplements<GenericNodeTransformer>()
export class ButtonControl {
    public static transform(node: ButtonControlNode) {
        return `<button id="${node.id}">${node.controlType}</button>`;
    }
}

export default ButtonControl;
