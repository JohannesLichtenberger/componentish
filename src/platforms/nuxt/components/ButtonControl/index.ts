import { indentation } from "../../../../generic/config/indentation";
import { staticImplements } from "../../../../generic/decorators/staticImplements";
import { ButtonControlNode } from "../../../../generic/interfaces/ComponentNodes/ButtonControl";
import { GenericNodeTransformer } from "../../../../generic/interfaces/transformer/GenericNodeTransformer";

@staticImplements<GenericNodeTransformer>()
export class ButtonControl {
    public static generateMarkup(node: ButtonControlNode, level: number = 0) {
        const indent = indentation.repeat(level);
        return `${indent}<button id="${node.id}">${node.controlType}</button>\n`;
    }
    public static getImports() {
        return ["ButtonControl"];
    }
}

export default ButtonControl;
