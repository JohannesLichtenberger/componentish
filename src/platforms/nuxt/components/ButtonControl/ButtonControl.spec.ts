import { ButtonControlNode } from "../../../../generic/interfaces/ComponentNodes/ButtonControl";
import { ButtonControl } from "../ButtonControl";

const mockNode = {
    controlType: "ActionControl",
    id: "ID_ActionControl_1_1",
    route: {
        path: "/viewpageBD246FF2",
    },
};
test("ButtonControl Snapshot", () => {
    const button = ButtonControl.generateMarkup(mockNode as ButtonControlNode);
    expect(button).toMatchSnapshot();
});
