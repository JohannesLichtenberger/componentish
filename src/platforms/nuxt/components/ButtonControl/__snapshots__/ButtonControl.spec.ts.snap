// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonControl Snapshot 1`] = `
"<button id=\\"ID_ActionControl_1_1\\">ActionControl</button>
"
`;
